### PR TITLE
Remove heatmap and display pie charts for confusion matrix

### LINKED ISSUE #2427: [FEAT] improve confusion matrix display
1. the percentages looks weird (should sum to 100)
2. add a column containing info of the accuracy per class

### PR DESCRIPTION
**Pull request by harsh-deepchecks**
#### Reference Issues/PRs

Fixes #2427 and [DEE-439 - "[FEAT] improve confusion matrix display" (deepchecks/deepchecks #2427)](https://linear.app/deepchecks/issue/DEE-439/[feat]-improve-confusion-matrix-display-deepchecksdeepchecks-2427)

#### What does this implement/fix? Explain your changes.
Removed the heatmap and added following information in the display:
- Added a piece of text displaying the overall accuracy of the model (by summing up all the accuracies and diving by the number of labels) and other relevant stats.
- After that, we display a max of 3 pie charts denoting the worst performing classes. Note that, we only consider the wrong predicted classes. We will consider the top three least accuracy classes and then will display a maximum of 3 pie charts based on the number of worst classes found. On hover on these pies, it displays the label name, the number of samples which are detected as the hovered label. Also, there is a threshold upto which the number of labels will be displayed in the pie chart. Above which, we will display as "Others" category. However, on hover, we display a comma separated list of label names for "Others"
- Added the relevant test cases for vision, NLP, tabular confusion matrix report.

#### Any other comments?
Its an initial PR for everyone to review the below displays and provide feedback.
- For NLP confusion matrix:
    - image

- For Tabular confusion matrix:
    - If all classes have accuracy of 100%: 
         -  image
    - If all classes does not have accuracy of 100%:
        - image

- For Vision confusion matrix:
      -  image

---
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/deepchecks/deepchecks/blob/main/CONTRIBUTING.rst

### CODE REVIEW

File: deepchecks/utils/abstracts/confusion_matrix_abstract.py [hunk 1]
```diff
@@ -9,40 +9,46 @@
...
 from deepchecks import ConditionCategory, ConditionResult
 from deepchecks.core import CheckResult
 from deepchecks.core.errors import DeepchecksValueError
-from deepchecks.utils.strings import format_number_if_not_nan, format_percent
+from deepchecks.utils.strings import format_percent
 
 __all__ = ['create_confusion_matrix_figure', 'run_confusion_matrix_check']
 
+MAX_WORST_CLASSES_TO_DISPLAY = 3
+MAX_TOP_CLASSES_TO_DISPLAY = 3
```
Comment (Nadav-Barak) [inline deepchecks/utils/abstracts/confusion_matrix_abstract.py:29]: Redundant? at least i couldn't find where it is used

File: deepchecks/utils/abstracts/confusion_matrix_abstract.py [hunk 2]
```diff
@@ -51,38 +57,69 @@ def create_confusion_matrix_figure(confusion_matrix_data: np.ndarray, classes_na
...
+    display_msg = textwrap.dedent(
+        f'The overall accuracy of your model is: {round(np.sum(accuracy_array)/len(accuracy_array), 2)}%.<br>'
+        f'Best accuracy achieved on samples with <b>{classes_names[np.argmax(accuracy_array)]}</b> '
+        f'label ({np.max(accuracy_array)}%).<br>Worst accuracy achieved on samples with '
+        f'<b>{classes_names[np.argmin(accuracy_array)]}</b> label ({np.min(accuracy_array)}%).<br>'
+        'Below are pie charts showing the prediction distribution for samples grouped based on their label.'
+    )
+    display.append(display_msg)
+    curr_col = 1
+    if len(worst_class_indices) > 0:
```
Comment (Nadav-Barak) [inline deepchecks/utils/abstracts/confusion_matrix_abstract.py:91]: would revert the if - check if len(worst_class_indices) == 0 then only return the text + would also return the first line of the text ( without the Worst accuracy achieved ...)

File: deepchecks/utils/abstracts/confusion_matrix_abstract.py [hunk 3]
```diff
@@ -51,38 +57,69 @@ def create_confusion_matrix_figure(confusion_matrix_data: np.ndarray, classes_na
...
+                values = values[sorted_indices]
+                labels = labels[sorted_indices]
+                other_labels_sum = np.sum(values[max_num_labels_to_show:])
+                other_label_names = ', '.join(labels[max_num_labels_to_show:])
+                labels = labels[:max_num_labels_to_show]
+                values = values[:max_num_labels_to_show]
+                values = np.append(values, other_labels_sum)
+                labels = np.append(labels, 'Others')
+
+            fig.add_trace(go.Pie(values=values, title=f'Label: <b>{classes_names[idx]}</b><br>'
```
Comment (Nadav-Barak) [inline deepchecks/utils/abstracts/confusion_matrix_abstract.py:111]: Would make the header for each plot:
** Anger ** (can use html commends like header / sub-header)
'# Samples': X, 'Accurate predictions':  Y (Z%) (try same line, liked the bold you did on the values)

File: deepchecks/utils/abstracts/confusion_matrix_abstract.py [hunk 4]
```diff
@@ -51,38 +57,69 @@ def create_confusion_matrix_figure(confusion_matrix_data: np.ndarray, classes_na
...
+
+    if len(accuracy_array) > MAX_WORST_CLASSES_TO_DISPLAY:
+        worst_class_indices = worst_class_indices[:MAX_WORST_CLASSES_TO_DISPLAY]
+
+    display_msg = textwrap.dedent(
+        f'The overall accuracy of your model is: {round(np.sum(accuracy_array)/len(accuracy_array), 2)}%.<br>'
+        f'Best accuracy achieved on samples with <b>{classes_names[np.argmax(accuracy_array)]}</b> '
+        f'label ({np.max(accuracy_array)}%).<br>Worst accuracy achieved on samples with '
+        f'<b>{classes_names[np.argmin(accuracy_array)]}</b> label ({np.min(accuracy_array)}%).<br>'
+        'Below are pie charts showing the prediction distribution for samples grouped based on their label.'
```
Comment (Nadav-Barak) [inline deepchecks/utils/abstracts/confusion_matrix_abstract.py:87]: On second thought IMO remove this sentance

**Review by Nadav-Barak — COMMENTED** [review]
Looks good in general, small comments - please also verify manually on 1-2 examples that the values displayed are correct

File: deepchecks/utils/abstracts/confusion_matrix_abstract.py (hunk 1 above)
Comment (harsh-deepchecks) [reply deepchecks/utils/abstracts/confusion_matrix_abstract.py]: Done

File: deepchecks/utils/abstracts/confusion_matrix_abstract.py (hunk 4 above)
Comment (harsh-deepchecks) [reply deepchecks/utils/abstracts/confusion_matrix_abstract.py]: Done

File: deepchecks/utils/abstracts/confusion_matrix_abstract.py (hunk 2 above)
Comment (harsh-deepchecks) [reply deepchecks/utils/abstracts/confusion_matrix_abstract.py]: Done

**Comment by harsh-deepchecks** [comment]
Closing as discussed, heatmaps are required. Will create a separate PR.

**Pull request closed without merging**